src/rgw/rgw_sal_motr.cc

```diff
@@ -685,6 +691,12 @@ int MotrBucket::put_info(const DoutPrefixProvider *dpp, bool exclusive, ceph::re
 {
   bufferlist bl;
   struct MotrBucketInfo mbinfo;
+  string tenant_bucket = info.bucket.tenant;
+  if (tenant_bucket != "") {
+    tenant_bucket += "$"+info.bucket.name;
+  } else {
+    tenant_bucket = info.bucket.name;
+  }
```

Comment:
This code block is repeated too many times. Better to create a function and call it with appropriate parameters.

Reply:
also make the function inline if possible.

Reply:
OR we can write a generic function to join with $ as well. whichever seems good. 😄 
```c++
// use
string bucket = join_strings("$", {tenant, bucket_name});

// definition
std::string join_strings(std::string connector, const std::vector<std::string>& segments) {
    string full_string = segments[0];
    for(int i = 1; i < segments.size(); i++)
        full_string += connector + segments[i];
    return full_string;
}
```

Reply:
OR we can write a generic utility function to join strings. Whichever looks good.. 😄
```c++
// Use
string bucket = join_strings("$", {tenant, bucket});

// Definition
std::string join_strings(std::string connector, const std::vector<std::string>& segments) {
  std::string full_string = "";
  for(int i = 0; i < segments.size(); i++) {
    if(segments[i] == "")
      continue;
    if(full_string != "")
      full_string += connector;
    full_string += segments[i];
  }
  return full_string;
}
```

Reply:
Great suggestion @sumedhak27 but it would be overkill in our case. Let's stick to '+' operator as of now.